ignite/contrib/handlers/trains_logger.py

```diff
@@ -0,0 +1,634 @@
+            )
+
+
+class GradsHistHandler(BaseWeightsHistHandler):
+    """Helper handler to log model's gradients as histograms.
+
+    Examples:
+
+        .. code-block:: python
+
+            from ignite.contrib.handlers.trains_logger import *
+
+            # Create a logger
+
+            trains_logger = TrainsLogger(project_name="pytorch-ignite-integration",
+                                         task_name="cnn-mnist"
+                                         )
+
+            # Attach the logger to the trainer to log model's weights norm after each iteration
+            trains_logger.attach(trainer,
+                                 log_handler=GradsHistHandler(model),
+                                 event_name=Events.ITERATION_COMPLETED)
+
+    Args:
+        model (torch.nn.Module): model to log weights
+        tag (str, optional): common title for all produced plots. For example, 'generator'
+
+    """
+
+    def __init__(self, model, tag=None):
+        super(GradsHistHandler, self).__init__(model, tag=tag)
+
+    def __call__(self, engine, logger, event_name):
+        if not isinstance(logger, TrainsLogger):
+            raise RuntimeError("Handler 'GradsHistHandler' works only with TrainsLogger")
+
+        global_step = engine.state.get_event_attrib_value(event_name)
+        tag_prefix = "{}/".format(self.tag) if self.tag else ""
+        for name, p in self.model.named_parameters():
+            if p.grad is None:
+                continue
+
+            title_name, _, series_name = name.partition(".")
+
+            logger.grad_helper.add_histogram(
+                title="{}grads_{}".format(tag_prefix, title_name),
+                series=series_name,
+                step=global_step,
+                hist_data=p.grad.detach().cpu().numpy(),
+            )
+
+
+class TrainsLogger(BaseLogger):
+    """
+    `Trains <https://github.com/allegroai/trains>`_ handler to log metrics, text, model/optimizer parameters,
+    plots during training and validation.
+    Also supports model checkpoints logging and upload to the storage solution of your choice (i.e. Trains File server,
+     S3 bucket etc.)
+
+    .. code-block:: bash
+
+        pip install trains
+        trains-init
+
+    Args:
+        project_name (str): The name of the project in which the experiment will be created. If the project
+            does not exist, it is created. If ``project_name`` is ``None``, the repository name is used. (Optional)
+        task_name (str): The name of Task (experiment). If ``task_name`` is ``None``, the Python experiment
+            script's file name is used. (Optional)
+        task_type (str): Optional. The task type. Valid values are:
+            - ``TaskTypes.training`` (Default)
+            - ``TaskTypes.train``
+            - ``TaskTypes.testing``
+            - ``TaskTypes.inference``
+        report_freq (int): Optional. Histogram processing frequency (handle hist values every X calls to the handler).
```

Comment:
@jkhenning I was checking documentation layout of `TrainsLogger` and seems like I missed those args:
- report_freq
- histogram_update_freq_multiplier

In ignite we can easily filter out how frequently call an event handler, so, does `report_freq` param mean to call `WeightsScalarHandler` every `report_freq` times :
```python
            trains_logger.attach(trainer,
                                 log_handler=WeightsScalarHandler(model),
                                 event_name=Events.ITERATION_COMPLETED(every=report_freq))
```
?

Reply:
Hi @vfdev-5 ,

I think you're right, and it was my mistake to expose these values (they are related to the internal working of how we process and average the reports internally). I'll open a PR and remove them, if you'd like.

Reply:
Well, if it does almost the same as event filtering, so, yes please send a PR and remove them...

Reply:
@vfdev-5 opened https://github.com/pytorch/ignite/pull/1036